run-build.sh

```diff
@@ -189,7 +189,7 @@ fi
 export DOTNET_SKIP_FIRST_TIME_EXPERIENCE=1
 
 if [ $BUILD -eq 1 ]; then
-    dotnet msbuild build.proj /p:Architecture=$ARCHITECTURE $CUSTOM_BUILD_ARGS /p:GeneratePropsFile=true /t:WriteDynamicPropsToStaticPropsFiles $argsnotargets
+    dotnet msbuild build.proj /p:Architecture=$ARCHITECTURE $CUSTOM_BUILD_ARGS /p:GeneratePropsFile=true /t:WriteDynamicPropsToStaticPropsFiles ${argsnotargets[@]}
```

Comment:
@peterhuene @johnbeisner In https://github.com/dotnet/source-build/pull/418#pullrequestreview-112111966 there was some question about if this was the right fix--just making sure this looks ok now before merging it into CLI proper.

Reply:
While this doesn't semantically change the script, I still think we're intending to pass all the arguments through. @johnbeisner can you confirm that is the original intent? If so, let's make the proper fix in this PR.

Reply:
It seems like you're saying that `$argsnotargets` is semantically the same as `${argsnotargets[@]}`, but this change does start passing all the arguments through. We needed that behavior change for source-build. Am I reading your comment right? Admittedly I don't know much about bash arrays (I tend to avoid them to stay more posixy).

Reply:
Hah, that's what I get for reviewing it on my phone.  I saw `0` and not `@`.

Comment withdrawn.  This change is correct.